workspaces/adventure-pack/goodies/java/src/integer_digits/AP.java

```diff
@@ -0,0 +1,30 @@
+package integer_digits;
+
+import iterable_int_stream.IterableIntStream;
+import java.util.stream.IntStream;
+
+public final class AP {
+
+  /**
+   * Get the digits in the given integer. For example, in the given integer 8725, digits are
+   * [8, 7, 2, 5].
+   * @param number Any number of interest.
+   * @return The {@link IterableIntStream stream} of digits in the given integer.
+   */
+  public static IterableIntStream integerDigits(int number) {
+    var digitsInReverse = IntStream.iterate(
+      number,
+      n1 -> n1 != 0,
+      n1 -> n1 / 10
+    )
+      .map(n1 -> n1 % 10)
+      .toArray();
+
+    var streamBuilder = IntStream.builder();
+    for (int i = digitsInReverse.length - 1; i >= 0; i--) {
```

Comment:
This makes me think if we could have a reverse function if one does not already exist.

Reply:
Nit: use pre-increment/pre-decrement in for loops unless post increment is necessary.

Reply:
> This makes me think if we could have a reverse function if one does not already exist.

You mean reverse method for IterableIntStream class, right?

Reply:
> Nit: use pre-increment/pre-decrement in for loops unless post increment is necessary.

Okay.

Reply:
Agreed, having a goody for reversing arrays would be nice! @utsavatwork if you're up for it, it would be great to have an issue created for this, or let us know if you'd prefer someone else create the issue.

Reply:
Oops, looks like we were commenting at the same time @utsavatwork. I'd personally be interested in a function for reversing an array. I know there's `Collections.reverse` for collections, but I don't think there's a `Arrays.reverse` yet?

But I'm not sure if that's what @volkmchl meant.

Reply:
> Agreed, having a goody for reversing arrays would be nice! @utsavatwork if you're up for it, it would be great to have an issue created for this, or let us know if you'd prefer someone else create the issue.

yeah, I'll create the issue.

> Oops, looks like we were commenting at the same time @utsavatwork. I'd personally be interested in a function for reversing an array. I know there's Collections.reverse for collections, but I don't think there's a Arrays.reverse yet?
But I'm not sure if that's what @volkmchl meant

hey, about the new goody for reversing the array,
1) Should I just write a static reverse() method under AP class to reverse the given generic array?  OR
2) Should it be a class, say 'CustomArrays' extending java.util.Arrays and then I can add reverse() method there?

Reply:
I think option 1 is fine for now. If we end up with too many things in the `AP` namespace, we can group things in a different way later.

Reply:
Created the issue https://github.com/code-chronicles-code/leetcode-curriculum/issues/367